extension/observer/cfgardenobserver/config_test.go

```diff
@@ -26,30 +27,181 @@ func TestLoadConfig(t *testing.T) {
+			expected: &Config{
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/garden.sock",
+				},
+				RefreshInterval:   1 * time.Minute,
+				CacheSyncInterval: 5 * time.Minute,
+				IncludeAppLabels:  true,
+				CloudFoundry: CfConfig{
+					Endpoint:     "https://api.cf.mydomain.com",
+					AuthType:     "token",
+					AccessToken:  "myaccesstoken",
+					RefreshToken: "myrefreshtoken",
+				},
 			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.id.String(), func(t *testing.T) {
-			cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
-			require.NoError(t, err)
-			factory := NewFactory()
-			cfg := factory.CreateDefaultConfig()
-			sub, err := cm.Sub(tt.id.String())
-			require.NoError(t, err)
-			require.NoError(t, sub.Unmarshal(cfg))
-
+			cfg := loadConfig(t, tt.id)
 			assert.NoError(t, component.ValidateConfig(cfg))
 			assert.Equal(t, tt.expected, cfg)
 		})
 	}
 }
+
+func TestConfigValidate(t *testing.T) {
+	cases := []struct {
+		reason string
+		cfg    Config
+		msg    string
+	}{
+		{
+			reason: "missing endpoint",
+			cfg: Config{
+				IncludeAppLabels: true,
+			},
+			msg: "config.Endpoint must be specified when include_app_labels is set to true",
+		},
+		{
+			reason: "missing auth_type",
+			cfg: Config{
+				IncludeAppLabels: true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+				},
+			},
+			msg: "config.authType must be specified when include_app_labels is set to true",
+		},
+		{
+			reason: "unknown auth_type",
+			cfg: Config{
+				IncludeAppLabels: true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: "unknown",
+				},
+			},
+			msg: "configuration option `auth_type` must be set to one of the following values: [user_pass, client_credentials, token]. Specified value: unknown",
+		},
+		{
+			reason: "missing username",
+			cfg: Config{
+				IncludeAppLabels: true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: authTypeUserPass,
+				},
+			},
+			msg: fieldError(authTypeUserPass, "username").Error(),
+		},
+		{
+			reason: "missing clientID",
+			cfg: Config{
+				IncludeAppLabels: true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: authTypeClientCredentials,
+				},
+			},
+			msg: fieldError(authTypeClientCredentials, "client_id").Error(),
+		},
+		{
+			reason: "missing AccessToken",
+			cfg: Config{
+				IncludeAppLabels: true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: authTypeToken,
+				},
+			},
+			msg: fieldError(authTypeToken, "access_token").Error(),
+		},
+	}
+
+	for _, tCase := range cases {
+		t.Run(tCase.reason, func(t *testing.T) {
+			err := tCase.cfg.Validate()
+			require.EqualError(t, err, tCase.msg)
+		})
+	}
+}
+
+func loadConf(t testing.TB, path string, id component.ID) *confmap.Conf {
```

Comment:
`loadConfig` and `loadConf` are a bit unclear on how they differ and their purpose. Can we rename to make their difference more clear? `loadConf` could be something like `loadConfFromFile`, or something else if you have a better idea.